storage/sqlite3/nodereputation.go

```diff
@@ -0,0 +1,447 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package nodereputation
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+
+	rep "storj.io/storj/pkg/nodereputation"
+	proto "storj.io/storj/protos/nodereputation"
+
+	// import of sqlite3 for side effects
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/zeebo/errs"
+)
+
+// CreateTableError is an error class for errors related to the reputation package
+var CreateTableError = errs.Class("reputation table creation error")
+
+// CreateNodeError is an error class for errors related to the reputation package
+var CreateNodeError = errs.Class("reputation node creation error")
+
+// SelectError is an error class for errors related to the reputation package
+var SelectError = errs.Class("reputation selection error")
+
+// UpdateError is an error class for errors related to the reputation package
+var UpdateError = errs.Class("reputation update error")
+
+// StartDBError is an error class for errors related to the reputation package
+var StartDBError = errs.Class("reputation start sqlite3 error")
+
+// StartDB starts a sqlite3 database from the file path parameter
+func StartDB(filePath string) (*sql.DB, error) {
+	db, err := sql.Open("sqlite3", filePath)
+	if err != nil {
+		return nil, StartDBError.Wrap(err)
```

Comment:
if we can't start the db, should we just exit the program? or write a method to try to try to open it again- after failed x times, close the program?